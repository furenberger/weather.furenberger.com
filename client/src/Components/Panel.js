import React from "react";
import {
  Container,
  Card,
  CardHeader,
  CardMedia,
  CardContent,
  TextField,
  Typography,
  Button
} from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  card: {
    padding: 15,
    marginTop: 24,
    marginBottom: 24,
    minWidth: 275,
  },
  cardHeader: {},
  cardContent: {},
  media: {
    height: 0,
    margin: 15,
    backgroundSize: "contain",
    backgroundRepeat: "no-repeat",
    paddingTop: "56.25%", // 16:9
  },
}));

const Panel = (props) => {
  const { selectedState } = props;
  const classes = useStyles();

  const renderStateCard = (selectedState) => {
    return (
      <React.Fragment>
        <Card className={classes.card}>
          <CardContent className={classes.cardContent}>
            <CardHeader
              className={classes.cardHeader}
              title={selectedState.stateName}
            />
            <CardMedia
              className={classes.media}
              image={selectedState.stateFlagUrl}
              title={selectedState.stateName + " state flag"}
            />
          </CardContent>
        </Card>
        <Card className={classes.card}>
          <CardContent className={classes.cardContent}>
            <CardHeader
              className={classes.cardHeader}
              title="Weather"
            />
            <TextField id="zipcode" required fullWidth label="Zip Code" autoComplete="billing postal-code"/>
            <Button variant="contained"
                      color="primary"
                      // onClick={this.handleNext}
                      >
                        Go!
                      </Button>
          </CardContent>
        </Card>
      </React.Fragment>
    );
  };

  const renderEmptyCard = () => {
    return (
      <Card className={classes.card}>
        <Typography variant="body2" color="textSecondary" component="p">
          Please select a state
        </Typography>
      </Card>
    );
  };

  return (
    <Container maxWidth="sm">
      {selectedState && selectedState.stateName
        ? renderStateCard(selectedState)
        : renderEmptyCard()}
    </Container>
  );
};

export default Panel;
