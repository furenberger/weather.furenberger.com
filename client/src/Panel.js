import React from "react";

const Panel = ({selectedState, setSelectedState}) => {
  return (
    <div className="panel">
      Selected State: {selectedState ? selectedState.stateName : ''}
    </div>
  );
};

export default Panel;
