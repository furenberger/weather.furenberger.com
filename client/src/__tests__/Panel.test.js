import React from "react";
import { render } from "@testing-library/react";
import Panel from "../Panel";

beforeEach(() => {
  jest.clearAllMocks();
});

describe("Panel.test", () => {
  it("renders without crashing", () => {
    const container = render(<Panel />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders empty card", () => {
    const { getByText } = render(<Panel />);
    expect(getByText("Please select a state")).toBeInTheDocument();
  });

  it("renders state card", () => {
    const selectedState = {
      stateName: "Wisconsin",
      id: "55",
      stateFlagUrl: 'http://flags.ox3.in/svg/us/wi.svg'
    };

    const { getByText } = render(<Panel selectedState={selectedState} />);
    expect(getByText("Wisconsin")).toBeInTheDocument();
  });
});
