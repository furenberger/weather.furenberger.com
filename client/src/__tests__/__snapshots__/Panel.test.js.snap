// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel.test renders without crashing 1`] = `undefined`;
